### PR TITLE
feat: add date and list branch operators

### PR DESCRIPTION
**Pull request by rita-gorokhod**
## What does this PR do?

Adds branch date operators (before, after, equals) and list (empty, not empty)
image

### CODE REVIEW

File: packages/engine/src/lib/handler/branch-executor.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export const branchExecutor: BaseExecutor<BranchAction> = {
 }
 
 
-function evaluateConditions(conditionGroups: BranchCondition[][]): boolean {
+export function evaluateConditions(conditionGroups: BranchCondition[][]): boolean {
```
Comment (rita-gorokhod) [inline packages/engine/src/lib/handler/branch-executor.ts:59]: For testing

**Comment by abuaboud** [comment]
Hi @rita-gorokhod 

Thank you for the contribution, I replaced the date comparison function with dayjs as they more readable and handle wider cases. 

It's now ready to merge once all checks pass.

**Review by abuaboud — APPROVED** [review]

**Comment by abuaboud** [comment]
Hi @rita-gorokhod 

We can ignore Validate publishable packages, Please sign the CLA and It would be ready to be merged

**Comment by rita-gorokhod** [comment]
Done @abuaboud thanks for the fixes!

**Review by abuaboud — APPROVED** [review]

**Pull request merged**